invenio_cli/helpers/commands.py

```diff
@@ -34,6 +35,10 @@ def build(self, pre, lock):
         """Delegates to build of environment commands."""
         return self.environment.build(pre, lock)
 
+    def __getattr__(self, name, **kwargs):
+        """Delegate commands according to environment."""
+        return getattr(self.environment, name, **kwargs)
+
```

Comment:
Neat! I had just changed to a classmethod `create` rather than an `__init__` to return the 2 different classes and stop repeating method signatures, but I wasn't too happy about how it didn't connect the 2 implementations. I will go that way.

We might as well get rid of the `build` method above and make sure it is always called with named arguments.